### PR TITLE
planner: fix wrong result when pushing Agg down through Union in MPP plans

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/45850

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
planner: fix wrong result when pushing Agg down through Union in MPP plans
```

### CODE REVIEW

File: planner/core/exhaust_physical_plans.go [hunk 1]
```diff
@@ -3215,6 +3215,16 @@ func (la *LogicalAggregation) tryToGetMppHashAggs(prop *property.PhysicalPropert
 	// Is this aggregate a final stage aggregate?
 	// Final agg can't be split into multi-stage aggregate
 	hasFinalAgg := len(la.AggFuncs) > 0 && la.AggFuncs[0].Mode == aggregation.FinalMode
+	// count final agg should become sum for MPP execution path.
+	// In the traditional case, TiDB take up the final agg role and push partial agg to TiKV,
+	// while TiDB can tell the partialMode and do the sum computation rather than counting but MPP doesn't
+	finalAggAdjust := func(aggFuncs []*aggregation.AggFuncDesc) {
+		for _, agg := range aggFuncs {
+			if agg.Mode == aggregation.FinalMode && agg.Name == ast.AggFuncCount {
+				agg.Name = ast.AggFuncSum
```
Comment (fixdb) [inline planner/core/exhaust_physical_plans.go:3224]: count() returns integer result type, does sum in TiFlash also return integer?

Comment (AilinKid) [reply planner/core/exhaust_physical_plans.go]: make sense, addressed

Comment (AilinKid) [reply planner/core/exhaust_physical_plans.go]: by the way, sum(arg) should return the same type with its args, now its arg come from sub-split-count, so the return type should be the same

**Review by fixdb — APPROVED** [review]
+1

**Review by qw4990 — APPROVED** [review]

**Pull request merged**